File: parent/739 -- Daily Temperatures/mine3.go
package main

func dailyTemperatures(temperatures []int) []int {
	var stack []int
	res := make([]int, len(temperatures))

	for i, temp := range temperatures {
		for len(stack) > 0 && temp > temperatures[stack[len(stack)-1]] {
			idx := stack[len(stack)-1]
			stack = stack[:len(stack)-1]
			res[idx] = i - idx
		}
		stack = append(stack, i)
	}

	return res
}

File: parent/739 -- Daily Temperatures/mine3.py
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        # Monotonic decreasing stack (Same as before, simpler)
        # Time: O(n)
        # Space: O(n) - Less space per stack entry
        res = [0] * len(temperatures)
        stack = []  # idx - Only store idx, since we can always retrieve temp from aray

        for i, temp in enumerate(temperatures):
            while stack and temp > temperatures[stack[-1]]:
                idx = stack.pop()
                res[idx] = i - idx

            stack.append(i)
            
        return res